tensorflow_datasets/scripts/cli/new_dataset.py

```diff
@@ -0,0 +1,256 @@
+    f.write(context.format(**data))
+
+
+def add_the_init(root_dir: str, data: Dict[str, str]) -> None:
+  """Creates a new __init__.py. file"""
+  file_path = os.path.join(root_dir, '__init__.py')
+
+  with tf.io.gfile.GFile(file_path.format(**data), 'w') as f:
+    f.write(_INIT_FILE.format(**data))
+
+
+def create_dataset_test_file(root_dir: str, data: Dict[str, str]) -> None:
+  """Create the test file associated with the dataset."""
+  file_path = os.path.join(root_dir, '{dataset_name}_test.py')
+  context = (_HEADER + _DATASET_TEST_DEFAULTS_IMPORTS + _DATASET_TEST_DEFAULTS)
+
+  with tf.io.gfile.GFile(file_path.format(**data), 'w') as f:
+    f.write(context.format(**data))
+
+
+def create_fake_data(root_dir: str, data: Dict[str, str]) -> None:
+  fake_examples_dir = os.path.join(root_dir, 'fake_examples')
+  fake_examples_dir = fake_examples_dir.format(**data)
+  tf.io.gfile.makedirs(fake_examples_dir)
+
+  fake_path = os.path.join(fake_examples_dir,
+                           'TODO-add_fake_data_in_this_directory.txt')
+  with tf.io.gfile.GFile(fake_path, 'w') as f:
+    f.write('{TODO}: Add fake data in this directory'.format(**data))
+
+
+def create_fake_data_gen_file(root_dir: str, data: Dict[str, str]) -> None:
+  fake_data_generator = os.path.join(root_dir, 'fake_data_generator.py')
+
+  content = (_HEADER + _FAKE_DATA_GEN_FILE)
+  with tf.io.gfile.GFile(fake_data_generator.format(**data), 'w') as f:
+    f.write(content.format(**data))
+
+
+def create_checksum_file(root_dir: str, data: Dict[str, str]) -> None:
+  checksum_path = os.path.join(root_dir, 'checksums.txt')
+  with tf.io.gfile.GFile(checksum_path.format(**data), 'w') as f:
+    f.write(_CHECKSUM_FILE.format(**data))
+
+
+def create_single_dataset(ds_name: str) -> None:
+  root_dir = os.path.abspath(ds_name)
+  tf.io.gfile.makedirs(os.path.join(ds_name, 'fake_examples'))
+
+  data = dict(
+      dataset_name=ds_name,
+      dataset_cls=naming.snake_to_camelcase(ds_name),
+      TODO='TODO({})'.format(ds_name),
+  )
+
+  create_dataset_file(root_dir, data)
+  add_the_init(root_dir, data)
+  create_dataset_test_file(root_dir, data)
+  create_fake_data(root_dir, data)
+  create_fake_data_gen_file(root_dir, data)
+  create_checksum_file(root_dir, data)
+
+  print(
+      'Dataset generated in {}\n'
+      'You can start with searching TODO({}).\n'
+      'Please check this '
+      '`https://github.com/tensorflow/datasets/blob/master/docs/add_dataset.md`'
+      'for details.'.format(root_dir, ds_name))
+
+
+def create_new_datasets(args: argparse.Namespace) -> None:
+  for ds_name in args.dataset_name:
+    create_single_dataset(ds_name)
+
+
+def add_new_dataset_parser(subparsers: argparse._SubParsersAction) -> None:  # pylint:disable = protected-access
+  """Add subparser for `new` command"""
+  new_ds_parser = subparsers.add_parser(
+      'new', help='Generated new dataset(s) files')
+
+  new_ds_parser.add_argument(
```

Comment:
I think it would be helpful to add a `--dst_dir` flag of type `pathlib.Path` which default to Path.cwd()